designs/2018-processors-improvements/README.md

```diff
@@ -0,0 +1,271 @@
+        ecmaVersion: 2018
+    },
+    overrides: [
+        {
+            files: ["*.md"],
+            processors: ["markdown/markdown", "vue/vue"]
+        }
+    ]
+};
+```
+
+This configuration applies two named processors to all files matching `*.md`. First, the `markdown` named processor from `eslint-plugin-markdown` is applied. The result of that preprocessing is then passed to the `vue` named processor from `eslint-plugin-vue`. The result of that preprocessing is then passed to ESLint for linting. When ESLint has completed linting, the reverse sequence occurs: the linting results are first passed to the `vue` named processor from `eslint-plugin-vue`, and once that postprocessing is complete, the result is then passed to the `markdown` named processor from  `eslint-plugin-markdown`.
+
+End-users can also mix and match processors, such as in this example:
+
+```js
+module.exports = {
+    plugins: ["markdown", "vue", "html"],
+    parserOptions: {
+        ecmaVersion: 2018
+    },
+    overrides: [
+        {
+            files: ["*.md"],
+            processors: ["markdown/markdown", "vue/vue"]
+        },
+        {
+            files: ["*.htm", "*.html"],
+            processors: ["html/html", "vue/vue"]
+        }
+    ]
+};
+```
+
+This configuration uses the same `vue` named processor from `eslint-plugin-vue` for both HTML and Markdown files, but only after an appropriate named processor is used to extract the Vue code.
+
+**Note:** It would be update to the Vue processor to understand the difference between different types of code it may be passed.
+
+### Virtual Filenames for Code Blocks
+
+The `preprocess()` method inside of a processor can now return an array containing either strings (the current behavior) or an object with two properties:
+
+* `filePath` - (string) a virtual file path to map the code block to
+* `text` - (string) the source code text
+
+For example:
+
+```js
+module.exports = {
+    preprocess(text, filename) {
+        
+        const {vueText, jsText } = doSomething(text, filename);
+        
+        return [
+            {
+                filePath: filename + ".vue",
+                text: vueText
+            },
+            {
+                filePath: filename + ".js",
+                text: jsText
+            }
+        ]
+    }
+};
+```
+
+This processor returns two code blocks: one that contains Vue code and is given a virtual filename ending with `.vue` and one that contains regular JavaScript code and is given a virtual filename ending with `.js`.
+
+When a `preprocess()` method returns an object with a `filePath` property, `CLIEngine` will call `getConfigForFile()` on the `filePath` property to determine the correct configuration for the code block.
+
+When a `preprocess()` method returns only a string, `CLIEngine` will interpret that as a JavaScript file and call `getConfigForFile()` using the parent file's filename to determine the correction configuration for the code block.
```

Comment:
I don't think that this process is needed because the configuration file will never change in this case.

Reply:
The file doesn't change, but the `overrides` needs to be applied for each piece coming from a processor. That happens inside of `getConfigForFile()` right now, so calling `getConfigForFile()` for the parent file and for a code block with a different extension might return different configs.

Reply:
Yes. I will make that clearer in the text.